### PR TITLE
418 review page privacy checkbox

### PR DESCRIPTION
**Pull request by nickjg231**
## Description
This PR adds the description prop and enables the JSX description option on the CheckboxField component. It also adds the privacy agreement checkbox to the burial form's review page.

## Original issue(s)
department-of-veterans-affairs/va-forms-system-core#0418

## Screenshots
image
image

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link to the original github issue has been provided
- [ ] No sensitive information (i.e. PII/credentials/internal URLs etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: examples/burial-form/ReviewPage.jsx [hunk 1]
```diff
@@ -440,10 +439,37 @@ export default function ReviewPage(props) {
               <Link to={page.pageUrl+'?edit=true&source='+page.id} className='vads-u-margin-bottom--1p5 review-page--page-heading--link'>Edit</Link>
             </div>
 
-            {bufferFields(page.fields)}
-          </section>
+                {bufferFields(page.fields)}
+              </section>
```
Comment (obliviga) [inline examples/burial-form/ReviewPage.jsx:443]: It looks like the indentation here may be a bit off here?

File: examples/burial-form/ReviewPage.jsx [hunk 2]
```diff
@@ -440,10 +439,37 @@ export default function ReviewPage(props) {
...
+          <CheckboxField
+            required="You must accept the privacy policy before continuing."
+            name="privacyAgreementAccepted"
+            label="I have read and accept the privacy policy"
+            description={null}
+          >
+            <p slot="description">
+              Please read and accept the{' '}
+              <a
+                aria-label="Privacy policy, will open in new tab"
```
Comment (obliviga) [inline examples/burial-form/ReviewPage.jsx:464]: Appreciate the helpful aria-label here!

File: examples/burial-form/ReviewPage.jsx [hunk 3]
```diff
@@ -440,10 +439,37 @@ export default function ReviewPage(props) {
...
-          </section>
+                {bufferFields(page.fields)}
+              </section>
+            )
+          }
         )}
-        ) }
+
+        <div>
+          <div className="vads-u-padding-y--2">
```
Comment (obliviga) [inline examples/burial-form/ReviewPage.jsx:449]: It might be more accessible to replace this `div` with a `p`, since there's a sentence here.

**Review by obliviga — APPROVED** [review]

**Comment by bjmccotter7192** [comment]
Once you merge main, could you update your package.json version and since you have a `feat` commit message i think the number would be 1.2.1.

**Review by bjmccotter7192 — APPROVED** [review]
Good looking out on the package.json and good work on the TSDocs 👍

**Pull request merged**